src/consensus/pbft/libbyz/Client_proxy.h

```diff
@@ -157,8 +181,9 @@ bool ClientProxy<T, C>::send_request(
   C* owner,
   bool is_read_only)
 {
-  if (out_reqs.size() >= Max_outstanding)
+  if (current_outstanding.fetch_add(1) >= Max_outstanding)
```

Comment:
I think this should be `current_outstanding.load()` so that we can remove the multiple `fetch_sub(1)` and only `fetch_add(1)` if we've successfully sent the request.